### PR TITLE
Fixes from 5.5

### PR DESCRIPTION
**Pull request by vtaing10**
## Description:
-Introduced floor-switch mode selection (Stairs, Elevator, Escalator) prior to route creation
-Automatically enforces elevator routing when wheelchair mode is enabled
-Restored indoor amenity/POI icons on maps
-Connected wheelchair routing to shared app settings and UI

## Related User Story:
5.4, 5.5, 5.7
## Screenshot/Demo:
image
## How to test:
Run the app, indoor navigation and choose 2 rooms (diff floors for options)
If wheelchair: locked on elevators

## Type of Change:
- [ ] Feature
- [ ] Bug fix
- [ ] Refactor
- [ ] Documentation update
- [ ] Tests
- [ ] Others

## Pull Request Checklist
- [ ] Flutter CI test passed
- [ ] Sonar test passed
- [ ] Codecov test passed
- [ ] Application runs without errors
- [ ] Feature tested locally on a high level
- [ ] Architecture conventions respected
- [ ] No API keys or secrets committed
- [ ] Document when necessary

## PR Review Checklist (dev)
- [ ] Code is readable and well structured
- [ ] No obvious bugs or edge cases missed
- [ ] Tests are adequate
- [ ] Naming and organization are clear
- [ ] Feature works as described
- [ ] No unnecessary complexity
- [ ] Follows project architecture

## PR Review Checklist (QA)
- [ ] Made sure to test the application fully
- [ ] No bugs created by this PR

### CODE REVIEW

File: lib/services/indoors_routing/indoor_multi_floor_router.dart [hunk 1]
```diff
@@ -80,6 +82,7 @@ class IndoorMultiFloorRouter {
   IndoorRoutePlan? _buildDifferentFloorRoute({
```
Comment (enyc24) [inline lib/services/indoors_routing/indoor_multi_floor_router.dart]: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. Use helper methods to make it less complex.

File: lib/shared/widgets/rooms_field_section.dart [hunk 2]
```diff
@@ -266,29 +297,136 @@ class _RoomFieldsSectionState extends State<RoomFieldsSection> {
...
+    return ChoiceChip(
+      label: Text(label),
+      selected: selected,
+      onSelected: onSelected == null ? null : (_) => onSelected(),
+      labelStyle: TextStyle(
+        color: selected
+            ? selectedForeground
+            : onSelected == null
+            ? unselectedForeground.withValues(alpha: 0.45)
+            : unselectedForeground,
```
Comment (enyc24) [inline lib/shared/widgets/rooms_field_section.dart]: Extract this nested ternary operation into an independent statement. This is complex, use independent statements to make it more simple for people to understand.

File: lib/shared/widgets/rooms_field_section.dart [hunk 3]
```diff
@@ -266,29 +297,136 @@ class _RoomFieldsSectionState extends State<RoomFieldsSection> {
...
+      selectedColor: selectedBackground,
+      backgroundColor: onSelected == null
+          ? disabledBackground
+          : unselectedBackground,
+      side: BorderSide(
+        color: selected
+            ? selectedForeground
+            : onSelected == null
+            ? unselectedForeground.withValues(alpha: 0.2)
+            : unselectedForeground.withValues(alpha: 0.7),
```
Comment (enyc24) [inline lib/shared/widgets/rooms_field_section.dart]: Extract this nested ternary operation into an independent statement.

**Comment by Cherryaki1** [comment]
Do note that the app crashed for a minute before generating the route from one floor to another, we should aim to optimize performance
